### PR TITLE
test: migrate utils tests to vitest

### PR DESCRIPTION
**Pull request by nazarhussain**
**Motivation**

Consolidate the testing frameworks and migrate to vitest.

**Description**

Migrate `utils` and spec tests to vitest.

**Steps to test or reproduce**

- Run all tests

### CODE REVIEW

File: packages/utils/test/unit/sleep.test.ts [hunk 1]
```diff
@@ -13,20 +12,19 @@ describe("sleep", function () {
     const controller = new AbortController();
     setTimeout(() => controller.abort(), 10);
 
-    // Sleep for longer than the current test timeout.
-    // If the abort signal doesn't work mocha will throw a timeout error
```
Comment (nflaig) [inline packages/utils/test/unit/sleep.test.ts:17]: Comment seems relevant

File: packages/utils/test/unit/timeout.test.ts [hunk 2]
```diff
@@ -1,14 +1,11 @@
...
-import {expect} from "chai";
+import {describe, it, expect, afterEach} from "vitest";
 import {withTimeout} from "../../src/timeout.js";
 import {ErrorAborted, TimeoutError} from "../../src/errors.js";
 
 describe("withTimeout", function () {
   const data = "DATA";
   const shortTimeoutMs = 10;
-  // Sleep for longer than the current test timeout.
-  // If the abort signal doesn't work mocha will throw a timeout error
```
Comment (nflaig) [inline packages/utils/test/unit/timeout.test.ts:10]: same as above

File: packages/utils/test/unit/sleep.test.ts (hunk 1 above)
Comment (nazarhussain) [reply packages/utils/test/unit/sleep.test.ts:17]: First it's not mocha any more, secondly the tests is clear that we are expecting an AbortError from the AbortController.

Comment (nflaig) [reply packages/utils/test/unit/sleep.test.ts:17]: > First it's not mocha any more

Yeah of course you would have to rephrase to vitest

>  clear that we are expecting an AbortError from the AbortController.

That is clear but why choose value 5000ms? seems random, sleep time only has to be sufficiently greater than timeout when abort signal is called

But not worth discussing, let's remove comment then

**Review by nflaig — APPROVED** [review]

**Comment by nazarhussain** [comment]
> That is clear but why choose value 5000ms?

Any value greater than the timeout value will have worked fine.

**Comment by wemeetagain** [comment]
:tada: This PR is included in [v1.14.0](https://github.com/ChainSafe/lodestar/releases/tag/v1.14.0) :tada:

**Pull request merged**